### PR TITLE
Make the example client more clear for those learning the crate.

### PR DESCRIPTION
**Pull request by shaleh**
Add explicit reading of the Response contents into a string.
Add more comments to be more clear for those learning the crate.

### CODE REVIEW

**Comment by GitCop** [comment]
Thanks for contributing! Unfortunately, I'm here to tell you there were the following style issues with your Pull Request:

 * Commit: eea3c2b4a366a7a5f3bda3ceadeeafec44d7a3de
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/hyperium/hyper/blob/master/CONTRIBUTING.md

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Thanks for contributing! Unfortunately, I'm here to tell you there were the following style issues with your Pull Request:

 * Commit: eea3c2b4a366a7a5f3bda3ceadeeafec44d7a3de
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/hyperium/hyper/blob/master/CONTRIBUTING.md

---
This message was auto-generated by https://gitcop.com

**Comment by shaleh** [comment]
What is wrong with my commit message? Is it an unknown scope?

**Comment by seanmonstar** [comment]
@shaleh no, the pull request still contains the previous commit, as gitcop says, eea3c2b. Try `git rebase -i master`

File: examples/client.rs [hunk 1]
```diff
@@ -22,11 +30,28 @@ fn main() {
...
+     * Check res.status to find out if it was successful.
+     * Inspect the headers to see what data type was returned and then
+     * parse the response data stored in `contents`.
+     */
     println!("Response: {}", res.status);
     println!("Headers:\n{}", res.headers);
-    io::copy(&mut res, &mut io::stdout()).unwrap();
+
+    let mut contents = String::new();
+    res.read_to_string(&mut contents).unwrap();
```
Comment (seanmonstar) [inline examples/client.rs]: If showing users that we must deal with errors, this seems like a good place to. Running this script against `https://google.com` will trigger this error. For whatever reason, Google doesn't return utf-8 if it doesn't recognize the user-agent.

File: examples/client.rs [hunk 2]
```diff
@@ -1,14 +1,22 @@
...
-use std::io;
+use std::io::{Read,};  // enables read_to_string
 
 use hyper::Client;
+use hyper::client::response::Response;
 use hyper::header::Connection;
 
+
+// add more parameters here to support adding custom headers
+fn process(client: &Client, url: &str) -> hyper::Result<Response> {
```
Comment (reem) [inline examples/client.rs]: I am not a fan of splitting this code out into a common function, in my opinion, the code is clearer and more idiomatic if we keep the use of `client` in a simple case like this down below, in main. I think clear variable names and comments will show examples of correct use.

**Comment by seanmonstar** [comment]
Thanks for the effort in this PR! I'm going to close this, as we're not quite sure if this makes the example easier to read, or rather, that it shows the parts we want it to.

**Pull request closed without merging**